central/cluster/datastore/datastore_impl.go

```diff
@@ -554,6 +556,10 @@ func (ds *datastoreImpl) postRemoveCluster(ctx context.Context, cluster *storage
 		log.Errorf("failed to delete external network graph entities for removed cluster %s: %v", cluster.GetId(), err)
 	}
 
+	if err := ds.scanSettingDatastore.RemoveClusterFromScanConfig(ctx, cluster.GetId()); err != nil {
```

Comment:
Wrap this in the feature flag because `ds.scanSettingDatastore` will be nil if the flag is off.